prow/gerrit/adapter/adapter.go

```diff
@@ -425,18 +426,43 @@ func (c *Controller) processChange(logger logrus.FieldLogger, instance string, c
 		labels      map[string]string
 		annotations map[string]string
 	}
-
 	var jobSpecs []jobSpec
-
-	changedFiles := client.ChangedFilesProvider(&change)
+	baseSHAGetter := func() (string, error) { return baseSHA, nil }
+	var hasRelatedChanges *bool
```

Comment:
should be inside headSHAGetter func?

Reply:
I intentionally left this outside the function so that the value would persist across multiple calls. This will avoid repeated calls to `HasRelatedChanges` and will instead reuse the result for subsequent calls.

Reply:
I see, so when `headSHAGetter` is invoked multiple times the same pointer will be updated